### PR TITLE
Add type support for values loaded from env vars, mainly to support $…

### PR DESCRIPTION
**Pull request by jonasfj**
…PORT

### CODE REVIEW

File: config/env/transform.go [hunk 1]
```diff
@@ -16,6 +30,47 @@ func init() {
 
 func (provider) Transform(cfg map[string]interface{}) error {
 	return config.ReplaceObjects(cfg, "env", func(val map[string]interface{}) (interface{}, error) {
-		return os.Getenv(val["$env"].(string)), nil
+		env := val["$env"].(string)
+		value := os.Getenv(env)
+		typ, ok := val["type"]
```
Comment (walac) [inline config/env/transform.go]: nit: `valueType` is a more descriptive variable name than `typ`.

File: config/env/transform.go [hunk 2]
```diff
@@ -1,9 +1,23 @@
 // Package configenv implements a TransformationProvider that replaces objects on
 // the form: {$env: "VAR"} with the value of the environment variable VAR.
+//
+// An additional key 'type' may be given as {$env: "VAR", type: "TYPE"}, which
+// determines how the environment variable value should be parsed. By default
+// the TYPE is 'string', but the following types are supported:
+//   - string, value is return as string (default),
+//   - number, value is parsed a number,
```
Comment (walac) [inline config/env/transform.go]: I think we should have different types for scalars and integers.

Comment (jonasfj) [inline config/env/transform.go]: No, the object hierarchy uses types matching what you get after `var t interface{}; json.Unmarshal(..., &t)`

So all integers will be loaded in as `float64`.. Consider these config transformers as functions that operate on JSON equivalent data structures:
 * `map[string]interface{}`
 * `interface{}`
 * `bool`
 * `float64`
 * `string`
 * `nil`

Maybe I should make some checks for that...

Comment (jonasfj) [inline config/env/transform.go]: Will add a recursive check to fix this later..

**Pull request merged**